### PR TITLE
Add more fields to wallet

### PR DESCRIPTION
**Pull request by lukechampine**
Add siafundOutputs, siacoinOutputs, and spentOutputs to the database.

Also improve unlock speed by determining transaction relevancy first.

### CODE REVIEW

File: modules/wallet/update.go [hunk 1]
```diff
@@ -4,54 +4,52 @@ import (
...
 
-	"github.com/NebulousLabs/Sia/build"
 	"github.com/NebulousLabs/Sia/modules"
 	"github.com/NebulousLabs/Sia/types"
 
 	"github.com/NebulousLabs/bolt"
 )
 
+// isWalletAddress is a helper function that an UnlockHash is derived from
+// one of the wallet's spendable keys.
```
Comment (DavidVorick) [inline modules/wallet/update.go]: confusing wording, I think you missed a word

**Comment by DavidVorick** [comment]
So far, so good. What work remains in getting instant wallet unlocking?

**Comment by DavidVorick** [comment]
Seems like you just need to be storing the processedTransactions? (next PR of course)

**Comment by lukechampine** [comment]
The fields in the current wallet persistence need to be moved over as well. And I need to write compatibility code for loading that persistence file into the db.

There's also the question of the seeds/keys: for security reasons, we can't store the keys in the database, correct? (Or could they be stored in encrypted form?)

**Comment by DavidVorick** [comment]
The seeds/keys is something that we can tackle in the next step. They might need to go in the database for the purposes of atomicity, but you'd just store them the way they are currently being stored on disk.

I'm thinking though that the keys should remain in-memory, because going to disk is really expensive and even if you've got 5000 keys they aren't going to consume that much RAM.

**Comment by lukechampine** [comment]
each `spendableKey` is going to be ~100 bytes, so 5000 keys would be <500kb. I worry about users who have many more keys than that, though.

**Pull request merged**